### PR TITLE
Added report for all the capital cities in the world organized by largest population to smallest and updated README.md with Requirements Met Table

### PR DESCRIPTION
**Pull request by Lindon-Jackasal**
Team,

This pull request contains the following:
-The report for all the capital cities in the world organised by largest population to smallest.
-Updated README.md file with the Project Requirements Met table to show all the requirements we need to meet or requirements we have already met. Please feel free to update the table as we progress.
-Added new java classes: CapitalCities.java, CapitalCitiesDAO interface, CapitalCitiesDAOImpl.java, CapitalCitiesReport.java & ReportCapitalCities.java

Kindly review and provide your feedback. If all is well, kindly approve and merge this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,3 +4,45 @@
 - develop Build Status [![Build Status](https://travis-ci.org/sousajf1/sem.svg?branch=develop)](https://travis-ci.org/sousajf1/sem)
 - [![LICENSE](https://img.shields.io/github/license/sousajf1/sem.svg?style=flat-square)](https://github.com/sousajf1/sem/blob/master/LICENSE)
 - [![Releases](https://img.shields.io/github/release/sousajf1/sem/all.svg?style=flat-square)](https://github.com/sousajf1/sem/releases)
```
Comment (sousajo-cc) [inline README.md:6]: looks good the table ty. Can you please also add the screenshot for the other already adressed cases?

File: src/main/java/com/napier/sem/App.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package com.napier.sem;
 
+import java.sql.SQLException;
```
Comment (sousajo-cc) [inline src/main/java/com/napier/sem/App.java]: is this import needed here?

File: src/main/java/com/napier/sem/App.java [hunk 3]
```diff
@@ -48,20 +52,44 @@ public void createReportFor(Region name, numberOfCountries topN) {
          * Create report.
          */
         public abstract void createReportFor();
+
```
Comment (sousajo-cc) [inline src/main/java/com/napier/sem/App.java]: sorry for being picky but avoid this pointless new lines, maybe it's your editor, it just adds noise to the review. here it's fine but in a big project if your editor does this all the time, people will just ask for changes in your PR right away to fix the newline mess without even taking a look at the logic of your changes :)

File: src/main/java/com/napier/sem/CapitalCitiesDAO.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * The interface Capital Cities dao.
+ */
+public interface CapitalCitiesDAO {
+    /**
+     * Gets all capital cities organized by largest population to smallest.
+     *
+     * @return all capital cities organized by largest population to smallest.
+     */
+    List<CapitalCities> getAllCapitalCities();
```
Comment (sousajo-cc) [inline src/main/java/com/napier/sem/CapitalCitiesDAO.java]: By the javadocs comment it seems that a more appropriate name would be `getAllCapitalCitiesPopFromLargestToSmallest`

File: src/main/java/com/napier/sem/App.java [hunk 5]
```diff
@@ -48,20 +52,44 @@ public void createReportFor(Region name, numberOfCountries topN) {
...
+    }
+
     /**
      * The entry point of application.
      *
      * @param args the input arguments
      */
     public static void main(String[] args) {
-        for(reportTypes t : reportTypes.values()){
+        for (reportTypes t : reportTypes.values()) {
```
Comment (sousajo-cc) [inline src/main/java/com/napier/sem/App.java]: please fix your editor, this is just whitespace noise :)

File: src/main/java/com/napier/sem/CapitalCitiesReport.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        CapitalCitiesDAO capitalcitiesDao = new CapitalCitiesDAOImpl();
+        List<CapitalCities> capitalcities = capitalcitiesDao.getAllCapitalCities();
+        for (CapitalCities c:capitalcities){
+            System.out.println(c.toString());
+        }
+    }
+
+    @Override
+    public void generateReport() {
+        capitalCitiesWorld();
```
Comment (sousajo-cc) [inline src/main/java/com/napier/sem/CapitalCitiesReport.java]: keeping the same naming convention as the other report should be fine here, so the method name can be `largestToSmallest`

Comment (sousajo-cc) [reply src/main/java/com/napier/sem/CapitalCitiesReport.java]: and maybe the `interface Report` could be reused

File: src/main/java/com/napier/sem/ReportCapitalCities.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package com.napier.sem;
+
+public interface ReportCapitalCities {
+
+    void capitalCitiesWorld();
+    void generateReport();
+}
```
Comment (sousajo-cc) [inline src/main/java/com/napier/sem/ReportCapitalCities.java]: can `interface Report `be re-used?

**Review by sousajo-cc — CHANGES_REQUESTED** [review]
looks good overall, few remarks, please take a look at those

**Comment by sousajo-cc** [comment]
please just do reset your local branch to `3cc2c1e296f20ac7936737b30deccdd959284b55`.
Everything there was fine except for the whitespaces and the `ReportCapitalCities` newly interface.
Just reset your branch to it and then reuse the `Report` interface instead of `ReportCapitalCities` that you created.

This newly changes are not good I'm sorry.

File: src/main/java/com/napier/sem/ReportTypes.java [hunk 8]
```diff
@@ -43,23 +43,38 @@ public void largestToSmallestPartial(numberOfCountries topN) {
...
 
     @Override
     public void generateReport(Region region) {
+        
         largestToSmallestInRegion(region);
     }
 
     @Override
     public void generateReport(numberOfCountries topN) {
+
```
Comment (sousajo-cc) [inline src/main/java/com/napier/sem/ReportTypes.java]: please don't add this white spaces. This makes the coded hard to read, and it's not consistent

**Comment by Lindon-Jackasal** [comment]
The changes are okay... I will explain

File: src/main/java/com/napier/sem/ReportTypes.java [hunk 9]
```diff
@@ -3,9 +3,9 @@
 import java.util.List;
 
 /**
- * The type Country report.
+ * The type of reports.
  */
-public class CountryReport implements Report {
+public class ReportTypes implements Report {
```
Comment (sousajo-cc) [inline src/main/java/com/napier/sem/ReportTypes.java]: why this? You can have a` countryReport` class and a `CapitalCitiesReport` both implementing the same interface.

**Comment by Lindon-Jackasal** [comment]
Hey Jorge,

I think I've made the requested changes. Please review.

**Review by sousajo-cc — DISMISSED** [review]

**Review by sousajo-cc — APPROVED** [review]
looks way better, thanks :D

**Pull request merged**